core/client/client_test.go

```diff
@@ -118,113 +119,155 @@ func TestPaginateHistory(t *testing.T) {
-					{Smr: &trillian.SignedMapRoot{MapRoot: []byte{0}}},
-					{Smr: &trillian.SignedMapRoot{MapRoot: []byte{1}}},
-					{Smr: &trillian.SignedMapRoot{MapRoot: []byte{2}}},
-					{Smr: &trillian.SignedMapRoot{MapRoot: []byte{3}}},
-					{Smr: &trillian.SignedMapRoot{MapRoot: []byte{4}}},
-				}},
-				{wantStart: 5, next: 10, items: []*pb.GetEntryResponse{
-					{Smr: &trillian.SignedMapRoot{MapRoot: []byte{5}}},
-					{Smr: &trillian.SignedMapRoot{MapRoot: []byte{6}}},
-					{Smr: &trillian.SignedMapRoot{MapRoot: []byte{7}}},
-					{Smr: &trillian.SignedMapRoot{MapRoot: []byte{8}}},
-					{Smr: &trillian.SignedMapRoot{MapRoot: []byte{9}}},
-				}},
-				{wantStart: 10, next: 0, items: []*pb.GetEntryResponse{
-					{Smr: &trillian.SignedMapRoot{MapRoot: []byte{10}}},
-				}},
+			wantValues: map[uint64][]byte{
+				0:  nil,
+				1:  nil,
+				2:  nil,
+				3:  nil,
+				4:  nil,
+				5:  nil,
+				6:  nil,
+				7:  nil,
+				8:  nil,
+				9:  nil,
+				10: nil,
 			},
 		},
 		{
 			desc:     "pageSize",
 			start:    0,
 			end:      5,
 			pageSize: 3,
-			reqs: []request{
-				// The value of next is opaque to the client.
-				{wantStart: 0, wantSize: 3, next: 1, items: []*pb.GetEntryResponse{
-					{Smr: &trillian.SignedMapRoot{MapRoot: []byte{0}}},
-					{Smr: &trillian.SignedMapRoot{MapRoot: []byte{1}}},
-					{Smr: &trillian.SignedMapRoot{MapRoot: []byte{2}}},
-					{Smr: &trillian.SignedMapRoot{MapRoot: []byte{3}}},
-				}},
-				{wantStart: 1, wantSize: 2, next: 0, items: []*pb.GetEntryResponse{
-					{Smr: &trillian.SignedMapRoot{MapRoot: []byte{5}}},
-					{Smr: &trillian.SignedMapRoot{MapRoot: []byte{6}}},
-				}},
+			wantValues: map[uint64][]byte{
+				0: nil,
+				1: nil,
+				2: nil,
+				3: nil,
+				4: nil,
+				5: nil,
 			},
 		},
 	} {
 		t.Run(tc.desc, func(t *testing.T) {
-			ctrl := gomock.NewController(t)
-			defer ctrl.Finish()
-			s, stop, err := testutil.NewMockKT(ctrl)
-			if err != nil {
-				t.Fatalf("NewMockKT(): %v", err)
-			}
-			defer stop()
-
 			c := Client{
 				Verifier: &fakeVerifier{},
 				cli:      s.Client,
 				pageSize: tc.pageSize,
 			}
 
-			for _, r := range tc.reqs {
-				s.Server.EXPECT().ListEntryHistory(gomock.Any(), matchers.ProtoEqual(&pb.ListEntryHistoryRequest{
-					AppId:    appID,
-					UserId:   userID,
-					Start:    r.wantStart,
-					PageSize: r.wantSize,
-				})).Return(&pb.ListEntryHistoryResponse{
-					NextStart: r.next,
-					Values:    r.items,
-				}, nil)
-			}
-
-			if _, _, err = c.PaginateHistory(ctx, appID, userID, tc.start, tc.end); err != tc.wantErr {
+			_, values, err := c.PaginateHistory(ctx, appID, userID, tc.start, tc.end)
+			if err != tc.wantErr {
 				t.Errorf("PaginateHistory(): %v, want %v", err, tc.wantErr)
 			}
+			if got, want := values, tc.wantValues; !reflect.DeepEqual(got, want) {
+				t.Errorf("PaginateHistory().values: \n%#v, want \n%#v, diff: \n%v",
+					got, want, pretty.Compare(got, want))
```

Comment:
(Optional)  Is there *any* chance at all that pretty.Compare and reflect.DeepEqual would return different results?  Might be worth using pretty.Compare for the initial comparison too, and checking the diff is empty of something?  Completely up to you though, as the two funcs will likely be in agreement the majority of the time.  Just a musing :)